custom_components/pyscript/trigger.py

```diff
@@ -816,31 +816,15 @@ async def trigger_watch(self):
                 #
                 # for state notify, check that changes occurred in state_trig_ident variables
                 #
-                if notify_type == 'state' and func_args['old_state'] is not None:
+                if notify_type == 'state':
                     trig_ident_changed = False
-                    changes = {}
-                    if func_args['old_state'].state != func_args['new_state'].state:
-                        changes[func_args['var_name']] = {
-                            "old": func_args['old_state'].state,
-                            "new": func_args['new_state'].state,
-                        }
-                        if func_args['var_name'] in self.state_trig_ident:
+                    for var in self.state_trig_ident:
+                        if new_vars.get(var) != new_vars.get(f"{var}.old"):
```

Comment:
You'll need to check if `var` is `domain.entity` (if so, just compare `value` with `old_value`) or `domain.entity.attr`(if so, compare `getattr(value, attr)` with `getattr(old_value, attr)`).